MIGRATION.md

```diff
@@ -0,0 +1,96 @@
+# Migration Guide
+
+This document describes the changes that need to be made to migrate from any of the 
+individual actions ([buf-setup-action][buf-setup], [buf-breaking-action][buf-breaking], [buf-lint-action][buf-lint], [buf-push-action][buf-push]).
+
+## [buf-setup-action][buf-setup]
+
+To migrate to `buf-action` while retaining the same behavior as the deprecated
+`buf-setup-action`, set the `setup_only` parameter to `true` and set the `version`
+parameter to the Buf CLI version you want to pin to. Here is an example:
+
+```diff
+ steps:
+   - uses: actions/checkout@v2
+-  - uses: bufbuild/buf-setup-action@v1.45.0
++  - uses: bufbuild/buf-action@v1
++    with:
++      version: '1.45.0' # Optional. Set the version, defaults to the latest release.
++      setup_only: true
+   - run: buf --version
+```
+
+The version of the `buf` CLI is no longer tied to the version of the action.
+See the setup-only example in the [README](./README.md#setup-only) for more
+details.
+
+## [buf-breaking-action][buf-breaking]
+
+To migrate from `buf-breaking-action` to `buf-action`, the `against` parameter
+has been renamed to `breaking_against` and will default to the head of the
+current branch of the pull request. In the following example, we replace with
+`buf-action` and remove the `against` parameter as it is the same as the default:
+
+```diff
+ steps:
+   - uses: actions/checkout@v2
+-  - uses: bufbuild/buf-setup-action@v1
+-  - uses: bufbuild/buf-breaking-action@v1
++  - uses: bufbuild/buf-action@v1
+     with:
+-      against: 'https://github.com/acme/weather.git#branch=main'
+```
+
+For more details on specifying inputs see the example in the 
+[README](./README.md#specify-the-input-directory).
+
+## [buf-lint-action][buf-lint]
+
+To migrate from `buf-lint-action` we can replace it with the default
+`buf-action` behaviour that runs `buf lint` on the input directory. No input
+changes are required. Here is an example:
+
+```diff
+ steps:
+   - uses: actions/checkout@v2
+-  - uses: bufbuild/buf-setup-action@v1
+-  - uses: bufbuild/buf-lint-action@v1
++  - uses: bufbuild/buf-action@v1
+     with:
+      input: proto
+```
+
+## [buf-push-action][buf-push]
+
+To migrate from `buf-push-action` to `buf-action`, there are a few changes that
+need to be made:
+  - `buf_token` has been renamed to `token`.
+  - `create_visibility` has been removed. If the BSR repository doesn't exist
+    it will always be created with the visbility of `private`. To disable
+    creation of repositories set the parameter `push_disable_create` to `true`. 
+  - `draft` has been removed. Drafts have been deprecated in favor of labels.
+    The default behavior of the action is to set the labels using git metadata.
+    This will be similar to the existing behaviour of drafts as the branch name
+    will be used for the label.
```

Comment:
If I am a user, this begs the question, what do I do if I want to customize the default behavior?

Reply:
I didn't want to repeat too much of the README but can add a section or link to the examples.

Reply:
I think something short here that:
1. makes it clear that the recommended path is to delete the old draft variable and use the default behavior
2. if you want to customize this behavior link to docs

Reply:
Now link to the git-metadata docs as part of the `drafts` removal description and at the bottom of the section to the commits and labels concept documentation.